module/zfs/dmu_object.c

```diff
@@ -31,6 +31,8 @@
 #include <sys/zap.h>
 #include <sys/zfeature.h>
 
+int zfs_metadnode_backfill = 0;
```

Comment:
I don't think that https://github.com/openzfs/openzfs/pull/82 is sufficient to address the performance problem that this patch is working around.  The problem is that dnode_next_offset (and dmu_object_next) don’t take into account objects allocated in memory but not yet synced to disk.  Therefore if we allocate more than a L1 (the comment about L2 is inaccurate) worth of dnodes in one txg, we will end up calling dnode_hold_impl / dmu_object_next on every allocated-in-memory object when cross the L1 boundary.